src/azure-cli-core/azure/cli/core/cloud.py

```diff
@@ -246,7 +246,7 @@ def _add_starting_dot(suffix):
 def _get_arm_endpoint(arm_dict, is_suffix=False):
     def _get_processed_arm_endpoint(name, add_dot=False, fallback_value=None):
         if is_suffix:
-            return (_add_starting_dot(arm_dict['suffixes'][name]) if add_dot else arm_dict['suffixes'][name]) if name in arm_dict['suffixes'] else fallback_value
+            return (_add_starting_dot(arm_dict['suffixes'][name]) if add_dot else arm_dict['suffixes'][name]) if 'suffixes' in arm_dict and name in arm_dict['suffixes'] else fallback_value
```

Comment:
If `arm_dict` does contain `suffixes` but `arm_dict['suffixes']` is `None`, `name in arm_dict['suffixes']` will fail:

```sh
arm_dict = {'suffixes': None}
name = 'test'
print('suffixes' in arm_dict and name in arm_dict['suffixes'])
```

Output:

```
Traceback (most recent call last):
  File "D:\cli\testproj\main.py", line 3, in <module>
    print('suffixes' in arm_dict and name in arm_dict['suffixes'])
                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: argument of type 'NoneType' is not iterable
```